src/agent/onefuzz/src/syncdir.rs

```diff
@@ -70,9 +70,14 @@ impl SyncedDir {
     pub async fn continuous_sync(
         &self,
         operation: SyncOperation,
-        delay: Option<Duration>,
+        delay_seconds: Option<u64>,
     ) -> Result<()> {
-        let delay = delay.unwrap_or(DEFAULT_CONTINUOUS_SYNC_DELAY);
+        let delay_seconds = delay_seconds.unwrap_or(DEFAULT_CONTINUOUS_SYNC_DELAY_SECONDS);
+        if delay_seconds == 0 {
```

Comment:
This is confusing. Without reading the implementation I would have expected that a delay of Some(0) continuously sync without any delay between syncs. 
I think the intent would be clearer if we don't call this function at all when sync is disabled

Reply:
The issue is backwards compatability.  Existing jobs don't have this defined, but instead don't have a value set and use ensemble.

We need to differentiate existing "no value set, therefore use ensemble with a 60s delay" from "no value set, disable ensemble".

I will add help to the sdk which will be printed during - -help that documents how to disable it.

Reply:
i was suggesting to do the check for the delay before we even call this function 
basically replacing calls like this
``` rust
let continuous_sync_task = inputs.continuous_sync(Pull, config.ensemble_sync_delay);
```
with something line 
``` rust
let continuous_sync_task = 
  if delay_seconds != 0{  
     inputs.continuous_sync(Pull, config.ensemble_sync_delay);
   } else {
    async{Ok()}
}
```
This makes it clearer at call site that we disable the sync when we receive 0.

Reply:
Ah, I understand.

My reason for putting it here was to reduce the number of places that the logic had to be specified.